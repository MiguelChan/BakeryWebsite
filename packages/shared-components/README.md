# `@mgl/shared-components`

> TODO: description

## Usage

```
const sharedComponents = require('@mgl/shared-components');

// TODO: DEMONSTRATE API
```
