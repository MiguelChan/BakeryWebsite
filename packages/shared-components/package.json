{
  "name": "@mgl/shared-components",
  "version": "1.1.0",
  "description": "The WebApp components library",
  "main": "./build/src/index.js",
  "types": "./build/src/index.d.ts",
  "files": [
    "./build/src/index.js"
  ],
  "scripts": {
    "clean": "rm -rf build",
    "build": "tsc",
    "test": "jest --config jest.config.js",
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint . --ext .ts --fix",
    "release": "npm run clean && npm run lint && npm run test && npm run build"
  },
  "devDependencies": {
    "eslint-plugin-modules-newline": "^0.0.6"
  },
  "peerDependencies": {}
}
