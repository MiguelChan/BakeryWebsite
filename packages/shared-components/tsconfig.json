{
    "compilerOptions": {
      "target": "ESNext",
      "module": "commonjs",
      "outDir": "./build",
      "declarationDir": "build",
      "declaration": true,
      "lib": ["ESNext"],
      "types": ["node", "jest"]
    },
    "include": [
      "src/**/*",
      "tst/**/*"
    ],
    "exclude": [
        "node_modules",
        "build"
    ]
  }