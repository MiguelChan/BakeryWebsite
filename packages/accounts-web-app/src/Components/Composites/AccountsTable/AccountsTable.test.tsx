import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { AccountType } from '@mgl/shared-components';
import {
  AccountsTable,
  AccountsTableProps,
} from './AccountsTable';

describe('AccountsTable', () => {
  const setupComponent = (args: AccountsTableProps): RenderResult => (
    render(<AccountsTable {...args} />)
  );

  it('Should display the AccountsTable', () => {
    const props: AccountsTableProps = {
      accounts: [
        {
          accountType: AccountType.Expenses,
          id: 'AnId',
          subAccounts: [],
          title: 'This is a title 1',
        },
        {
          accountType: AccountType.Capital,
          id: 'AnId2',
          subAccounts: [],
          title: 'This is a title 2',
        },
        {
          accountType: AccountType.Entry,
          id: 'AnId3',
          subAccounts: [],
          title: 'This is a title 3',
        },
      ],
    };

    const component = setupComponent(props);
    expect(component.container).toMatchSnapshot();
  });
});
