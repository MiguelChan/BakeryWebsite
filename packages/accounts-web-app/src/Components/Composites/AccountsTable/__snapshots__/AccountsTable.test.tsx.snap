// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsTable Should display the AccountsTable 1`] = `
<div>
  <table
    class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Id
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Nombre de cuenta
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
          scope="col"
        >
          Tipo de Cuenta
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          AnId
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          This is a title 1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Expenses
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          AnId2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          This is a title 2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Capital
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          AnId3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          This is a title 3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Entry
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
